docs/essentials/overriding.md

```diff
@@ -0,0 +1,151 @@
+# Overriding
+
+Overriding allows for the easy replacement of parts of SFX, thereby modifying, adding, or removing functionality. For example, you can easily override how adding to cart works, how header/footer looks, how the catalog is displayed etc.
+
+Overriding works in a file-based manner, so basically any file (page, component, service, composable, etc.) can be overridden.
+
+::: tip
+Overriding rules are managed in the `storefront-x.config.js` file based on the position of imported module. The file from a later module will be used instead of the file from an earlier module. If only one module is imported, the file from that module will be used.
+:::
+
+Overriding is one of the primary feature that allows Storefront X to be extremely flexible and performant at the same time.
+
+## How to override?
+
+Imagine we have a default `module-a` with the `Header.vue` component in the `/organisms` concept.
+
+By default, this header contains the default setup, but with `module-b` enabled, this header will be overridden to our setup. Let's imagine our company would like to have a fixed menu all the time while scrolling and a different color palette. The best way to achieve this is to override the component.
+
+```vue
+// module-a/organisms/Header.vue
+
+<template>
+  <header class="bg-white border-b-2 border-primary-400 fixed top-0 md:static z-20 left-0 right-0">
+    <HeaderLinks />
+    <HeaderMain />
+    <HeaderMenu />
+  </header>
+</template>
+
+<script setup lang="ts">
+import HeaderLinks from '#ioc/molecules/HeaderLinks'
+import HeaderMain from '#ioc/molecules/HeaderMain'
+import HeaderMenu from '#ioc/molecules/HeaderMenu'
+</script>
+```
+
+We can achieve it by creating the `organisms` concept (directory) inside a `module-b` and put a new file with the same name there.
+
+```vue
+// module-b/organisms/Header.vue
+
+<template>
+  <header class="bg-white fixed border-b-2 border-slate-500 top-0 z-20 left-0 right-0">
+    <section class="bg-slate-200">
+      <HeaderLinks />
+    </section>
+    <HeaderMain />
+    <HeaderMenu />
+  </header>
+</template>
+
+<script setup lang="ts">
+import HeaderLinks from '#ioc/molecules/HeaderLinks'
+import HeaderMain from '#ioc/molecules/HeaderMain'
+import HeaderMenu from '#ioc/molecules/HeaderMenu'
+</script>
+```
+
+Finally, we have to import `module-b` after `module-a` in the `storefront-x.config.js` file.
+
+```js
+// storefront-x.*.config.js
+
+export default {
+  modules: [
+    // some modules...
+    'module-a',
+    // some modules...
+    'module-b',
+    // some modules...
+  ],
+}
+```
+
+The component from `module-b` will be used instead of the component from `module-a`.
+
+Vice versa, if we import `module-b` before `module-a`, the component from `module-a` will be used. And if we import only one module (`module-a` or `module-b`), the component from that module will be used. **The modules are independent of each other.**
+
+### Nested files
```

Comment:
Just show nesting on components or other very primitive concept/directory. base/templates ideally shouldnt even be touched by 3rd party developers.